SerialPrograms/Source/PokemonSV/Options/PokemonSV_SandwichIngredientsTable.cpp

```diff
@@ -0,0 +1,72 @@
+/*  Sandwich Ingredients Table
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "PokemonSV/Resources/PokemonSV_Ingredients.h"
+#include "PokemonSV_SandwichIngredientsTable.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+
+
+SandwichIngredientsTableRow::SandwichIngredientsTableRow()
+    : item("lettuce")
+{
+    PA_ADD_OPTION(item);
+}
+std::unique_ptr<EditableTableRow> SandwichIngredientsTableRow::clone() const {
+    std::unique_ptr<SandwichIngredientsTableRow> ret(new SandwichIngredientsTableRow());
+    ret->item.set_by_index(item.index());
+    return ret;
+}
+
+SandwichIngredientsTable::SandwichIngredientsTable(std::string label)
+    : EditableTableOption_t<SandwichIngredientsTableRow>(
+        std::move(label),
+        LockWhileRunning::LOCKED,
+        make_defaults()
+        )
+{}
+
+
+bool SandwichIngredientsTable::find_item(const std::string& item_slug) const {
+    std::vector<std::unique_ptr<SandwichIngredientsTableRow>> table = copy_snapshot();
+    for (const std::unique_ptr<SandwichIngredientsTableRow>& row : table) {
+        if (row->item.slug() == item_slug) {
+            return true;
+        }
+    }
+    return false;
+}
+
+std::vector<std::string> SandwichIngredientsTable::selected_items() const {
```

Comment:
This is named "selected items", but it returns every slug in the table?

Reply:
Only items selected by the user ? it iterates over the table setuped by the user

Reply:
It ended up not being needed, so I removed it and one other function from the same file.